### PR TITLE
Basic validation of the entries through the lib

### PR DESCRIPTION
**Pull request by arturo-seijas**
Applicable spec: <link>
N/A

### Overview

Basic validation of the entries through the lib. Instead of the requirer data when valid, a tuple with the valid entries and the response for the invalid is returned

### Rationale

N/A

### Juju Events Changes

N/A

### Module Changes

N/A

### Library Changes

dns_record.py

### Checklist

- [ ] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [ ] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [ ] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [ ] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [ ] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by mthaddon — DISMISSED** [review]

File: lib/charms/bind/v0/dns_record.py [hunk 1]
```diff
@@ -201,7 +200,7 @@ class DNSRecordProviderData(BaseModel):
         dns_entries: list of entries to manage.
     """
 
-    dns_entries: List[DNSProviderData] = Field(min_length=1)
```
Comment (yanksyoon) [inline lib/charms/bind/v0/dns_record.py:204]: Out of curiosity, may I ask why the min_length is no longer applicable?

File: lib/charms/bind/v0/dns_record.py [hunk 2]
```diff
@@ -268,7 +266,8 @@ def validate_dns_entry(self, _: ValidationInfo) -> "RequirerEntry":
             the DNS entry if valid.
         """
         validated_entry = RequirerEntry.model_validate(self)
-        # Additional validations will be done here
+        # Additional validations will be done here in the form of assertions
+        # assert validated_entry.domain == "cloud.canonical.com"
```
Comment (yanksyoon) [inline lib/charms/bind/v0/dns_record.py:270]: Would this comment be intended?

File: lib/charms/bind/v0/dns_record.py [hunk 3]
```diff
@@ -347,40 +350,71 @@ def to_relation_data(self, model: ops.Model, relation: ops.Relation) -> Dict[str
...
+    ) -> Tuple["DNSRecordRequirerData", "DNSRecordProviderData"]:
         """Initialize a new instance of the DNSRecordRequirerData class from the relation data.
 
         Args:
             model: the Juju model.
             relation: the relation.
 
         Returns:
-            A DNSRecordRequirerData instance.
+            the DNSRecordRequirerData instance.
```
Comment (yanksyoon) [inline lib/charms/bind/v0/dns_record.py:363]: I think it might help explaining the tuple: (Valid DNSRecordRequirerData, Invalid DNSRecordProviderData) :D

**Review by yanksyoon — DISMISSED** [review]
LGTM! Minor suggestions :rocket: thank you!

File: lib/charms/bind/v0/dns_record.py (hunk 1 above)
Comment (arturo-seijas) [reply lib/charms/bind/v0/dns_record.py:204]: The resulting list might be empty if all validations fail

File: lib/charms/bind/v0/dns_record.py (hunk 2 above)
Comment (arturo-seijas) [reply lib/charms/bind/v0/dns_record.py:270]: Yes, for now. It will be removed later on

**Review by yanksyoon — APPROVED** [review]

**Review by mthaddon — APPROVED** [review]

**Pull request merged**